package.json

```diff
@@ -44,12 +44,18 @@
     "eslint": "^8.57.0",
     "eslint-config-next": "^15.0.0",
     "eslint-config-prettier": "^10.0.0",
+    "husky": "^9.1.7",
+    "lint-staged": "^15.5.0",
     "postcss": "^8.4.40",
     "prettier": "^3.3.3",
     "prettier-plugin-tailwindcss": "^0.6.5",
     "tailwindcss": "^3.4.7",
     "tailwindcss-logical": "^3.0.1",
     "typescript": "^5.5.4"
   },
-  "packageManager": "yarn@4.7.0"
-}
\ No newline at end of file
+  "packageManager": "yarn@4.7.0",
+  "lint-staged": {
+    "*": "prettier --ignore-unknown --write",
+    "*.{js,mjs,ts,tsx}": "lint --fix"
+  }
```

Comment:
I would prefer if we use a JS-based lint-staged config, with a separate step for `lint --fix` and typechecking (which can be `yarn typecheck` for consistency with the main repo).

Reply:
i pushed up a fix in 3a235f9, let me know if that's what you mean. note that i used `lint-js` instead of `lint:js`, as colons are [global scripts](https://yarnpkg.com/features/workspaces#global-scripts) for Yarn. that's less relevant here as it's not a workspace, but i think it'd be good to update that in the main repo.

Reply:
I dont really like having `lint-staged` run both at the same time, and using `&` in scripts is quite brittle (it will mess with outputs if both output at the same time).

We should use a JS config file for `lint-staged` because it allows to properly run them in parallel, and also to not have it pass a list of modified changes to `tsc`.

For example:

```
const config = {
  '*': 'prettier --ignore-unknown --write',
  '*.{js,jsx,mjs,ts,tsx}': 'eslint --fix',
  '*.ts?(x)': () => 'tsc -p tsconfig.json --noEmit',
```

Reply:
> using & in scripts is quite brittle (it will mess with outputs if both output at the same time).

not with Yarn, which [natively handles that pretty well](https://yarnpkg.com/features/scripting#parallel-tasks). but i can remove if you'd prefer not.

> We should use a JS config file for lint-staged because it allows to properly run them in parallel, and also to not have it pass a list of modified changes to tsc.

gotcha, so pretty much copy the main repo config for JS/TS files?